### ISSUE TITLE
NullPointerException in TokenRequestTask execution

### ISSUE THREAD

**Issue by trebouillon** [openid/AppAuth-Android#402]
I use the version 0.7.0 of the library and sometimes get the following error:

```
Caused by java.lang.NullPointerException
       at java.io.Reader.<init>(Reader.java:78)
       at java.io.InputStreamReader.<init>(InputStreamReader.java:72)
       at net.openid.appauth.Utils.readInputStream(Utils.java:36)
       at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:431)
       at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:375)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
```

This crash happens when the input stream is null, e.g. the server is out of business.

Would it be possible to handle a null input stream as network error?

**Comment by iainmcgin** [comment]
Yep, looks like a bug, will fix.

**Comment by trebouillon** [comment]
Hey, I already fixed the bug. I am going to contribute my fix as eBay contribution. Once the CLA is signed I will create a PR.

**Comment by iainmcgin** [comment]
Excellent! Looking forward to seeing the PR.

**Comment by trebouillon** [comment]
https://github.com/openid/AppAuth-Android/pull/414

**Comment by iainmcgin** [comment]
Should now be fixed by the merge of #414.

**Comment by 5eeman** [comment]
I've faced with the same issue or similar and can't find any solution at the moment.
```
java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.NullPointerException
        at java.io.Reader.<init>(Reader.java:78)
        at java.io.InputStreamReader.<init>(InputStreamReader.java:72)
        at net.openid.appauth.Utils.readInputStream(Utils.java:36)
        at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:453)
        at net.openid.appauth.AuthorizationService$TokenRequestTask.doInBackground(AuthorizationService.java:395)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```
